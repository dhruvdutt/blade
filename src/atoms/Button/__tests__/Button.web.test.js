import React from 'react';
import { fireEvent } from '@testing-library/react';

import Button from '../index';
import View from '../../View';
import { renderWithTheme } from '../../../_helpers/testing';

describe('<Button />', () => {
  describe('block', () => {
    it('renders block primary <Button />', () => {
      const { container, getByText } = renderWithTheme(<Button block>Click Me</Button>);
      expect(getByText(/click me/i)).toBeTruthy();
      expect(container).toMatchSnapshot();
    });
  });
  describe('size', () => {
    it('renders xsmall primary <Button />', () => {
      const { container, getByText } = renderWithTheme(<Button size="xsmall">Click Me</Button>);
      expect(getByText(/click me/i)).toBeTruthy();
      expect(container).toMatchSnapshot();
    });
    it('renders a small Primary <Button />', () => {
      const { container, getByText } = renderWithTheme(<Button size="small">Click Me</Button>);
      expect(getByText(/click me/i)).toBeTruthy();
      expect(container).toMatchSnapshot();
    });
    it('renders a small Primary <Button />', () => {
      const { container, getByText } = renderWithTheme(<Button size="small">Click Me</Button>);
      expect(getByText(/click me/i)).toBeTruthy();
      expect(container).toMatchSnapshot();
    });
    it('renders a large Primary <Button />', () => {
      const { container, getByText } = renderWithTheme(<Button size="large">Click Me</Button>);
      expect(getByText(/click me/i)).toBeTruthy();
      expect(container).toMatchSnapshot();
    });
  });
  describe('variant', () => {
    it('renders a small Primary <Button />', () => {
      const { container, getByText } = renderWithTheme(
        <Button variant="primary" size="small">
          Click Me
        </Button>,
      );
      expect(getByText(/click me/i)).toBeTruthy();
      expect(container).toMatchSnapshot();
    });
    it('renders a medium(Default) Secondary <Button />', () => {
      const { container, getByText } = renderWithTheme(
        <Button variant="secondary">Click Me</Button>,
      );
      expect(getByText(/click me/i)).toBeTruthy();
      expect(container).toMatchSnapshot();
    });
    it('renders a small Tertiary <Button />', () => {
      const { container, getByText } = renderWithTheme(
        <Button variant="tertiary" size="small">
          Click Me
        </Button>,
      );
      expect(getByText(/click me/i)).toBeTruthy();
      expect(container).toMatchSnapshot();
    });
    it('renders a large Secondary <Button />', () => {
      const { container, getByText } = renderWithTheme(
        <Button variant="secondary" size="large">
          Click Me
        </Button>,
      );
      expect(getByText(/click me/i)).toBeTruthy();
      expect(container).toMatchSnapshot();
    });
    it('renders a medium(Default) Tertiary <Button />', () => {
      const { container, getByText } = renderWithTheme(
        <Button variant="tertiary">Click Me</Button>,
      );
      expect(getByText(/click me/i)).toBeTruthy();
      expect(container).toMatchSnapshot();
    });
    it('Renders a large Tertiary <Button />', () => {
      const { container, getByText } = renderWithTheme(
        <Button variant="tertiary" size="large">
          Click Me
        </Button>,
      );
      expect(getByText(/click me/i)).toBeTruthy();
      expect(container).toMatchSnapshot();
    });
  });
  describe('disabled', () => {
    it('Renders a disabled Primary <Button />', () => {
      const { container, getByText } = renderWithTheme(<Button disabled>Click Me</Button>);
      expect(getByText(/click me/i)).toBeTruthy();
      expect(container).toMatchSnapshot();
    });
    it('renders a disabled Secondary <Button />', () => {
      const { container, getByText } = renderWithTheme(
        <Button disabled variant="secondary">
          Click Me
        </Button>,
      );
      expect(getByText(/click me/i)).toBeTruthy();
      expect(container).toMatchSnapshot();
    });
    it('Renders a disabled Tertiary <Button />', () => {
      const { container, getByText } = renderWithTheme(
        <Button disabled variant="tertiary">
          Click Me
        </Button>,
      );
      expect(getByText(/click me/i)).toBeTruthy();
      expect(container).toMatchSnapshot();
    });
  });
  describe('checks onClick functionality of <Button />', () => {
    it('checks onClick is called for <Button />', () => {
      const mockPress = jest.fn();
      const { getByText } = renderWithTheme(<Button onClick={mockPress}>Click Me</Button>);
      const button = getByText(/click me/i).parentNode;
      fireEvent.click(button);
      expect(mockPress).toBeCalledTimes(1);
    });
    it('checks onClick is not called for disabled <Button />', () => {
      const mockPress = jest.fn();
      const { getByText } = renderWithTheme(
        <Button disabled onClick={mockPress}>
          Click Me
        </Button>,
      );
      const button = getByText(/click me/i).parentNode;
      fireEvent.click(button);
      expect(mockPress).not.toBeCalled();
    });
  });
  describe('error', () => {
    beforeAll(() => jest.spyOn(console, 'error').mockImplementation());
    afterAll(() => jest.restoreAllMocks());
    it('throws error when <Button /> is not passed a string as children', () => {
      const errorMessage = 'Error in Button: expected `children` of type `string` but found object';
      expect(() =>
        renderWithTheme(
          <Button size="small">
            <View />
          </Button>,
        ),
      ).toThrow(errorMessage);
    });
  });
  describe('icon', () => {
    it('renders <Button /> with Info icon ', () => {
      const { container, getByText } = renderWithTheme(<Button icon="info">Click Me</Button>);
      expect(getByText(/click me/i)).toBeTruthy();
      expect(container).toMatchSnapshot();
    });
    it('renders a large Primary <Button /> with right icon', () => {
      const { container, getByText } = renderWithTheme(
        <Button size="large" icon="info" iconAlign="right">
          Click Me
        </Button>,
      );
      expect(getByText(/click me/i)).toBeTruthy();
      expect(container).toMatchSnapshot();
    });
    it('renders a large Primary <Button /> with left icon', () => {
      const { container, getByText } = renderWithTheme(
        <Button size="large" icon="info" iconAlign="left">
          Click Me
        </Button>,
      );
      expect(getByText(/click me/i)).toBeTruthy();
      expect(container).toMatchSnapshot();
    });
  });
});
