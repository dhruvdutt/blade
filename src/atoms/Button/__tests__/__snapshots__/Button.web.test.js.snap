// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> block renders block primary <Button /> 1`] = `
<div>
  <button
    class="Buttonweb__StyledButton-gxhxmy-0 gQgDyl Spaceweb__Space-njatzl-0 jGLucJ Sizeweb__Size-sc-10bl685-0 bPNzzG Flexweb__Flex-ni2cjg-0 hBDZyG"
    color="primary"
    data-testid="ds-button"
  >
    <div
      class="Textweb__Text-sc-6jf9e4-0 cJrlLI"
      color="light.100"
      data-testid="ds-text"
    >
      Click Me
    </div>
  </button>
</div>
`;

exports[`<Button /> disabled Renders a disabled Primary <Button /> 1`] = `
<div>
  <button
    class="Buttonweb__StyledButton-gxhxmy-0 kdwhVC Spaceweb__Space-njatzl-0 jGLucJ Sizeweb__Size-sc-10bl685-0 bPNzzG Flexweb__Flex-ni2cjg-0 dfbkqx"
    color="primary"
    data-testid="ds-button"
    disabled=""
  >
    <div
      class="Textweb__Text-sc-6jf9e4-0 cZfose"
      color="light.150"
      data-testid="ds-text"
    >
      Click Me
    </div>
  </button>
</div>
`;

exports[`<Button /> disabled Renders a disabled Secondary <Button /> 1`] = `
<div>
  <button
    class="Buttonweb__StyledButton-gxhxmy-0 ORQnb Spaceweb__Space-njatzl-0 jGLucJ Sizeweb__Size-sc-10bl685-0 bPNzzG Flexweb__Flex-ni2cjg-0 dfbkqx"
    color="primary"
    data-testid="ds-button"
    disabled=""
  >
    <div
      class="Textweb__Text-sc-6jf9e4-0 eGSpCE"
      color="shade.940"
      data-testid="ds-text"
    >
      Click Me
    </div>
  </button>
</div>
`;

exports[`<Button /> disabled Renders a disabled Tertiary <Button /> 1`] = `
<div>
  <button
    class="Buttonweb__StyledButton-gxhxmy-0 bMjOWC Spaceweb__Space-njatzl-0 jGLucJ Sizeweb__Size-sc-10bl685-0 bPNzzG Flexweb__Flex-ni2cjg-0 dfbkqx"
    color="primary"
    data-testid="ds-button"
    disabled=""
  >
    <div
      class="Textweb__Text-sc-6jf9e4-0 eGSpCE"
      color="shade.940"
      data-testid="ds-text"
    >
      Click Me
    </div>
  </button>
</div>
`;

exports[`<Button /> icon Renders a Primary <Button /> with left icon 1`] = `
<div>
  <button
    class="Buttonweb__StyledButton-gxhxmy-0 cXFeAh Spaceweb__Space-njatzl-0 jGLucJ Sizeweb__Size-sc-10bl685-0 bPNzzG Flexweb__Flex-ni2cjg-0 dfbkqx"
    color="primary"
    data-testid="ds-button"
  >
    <div
      class="Textweb__Text-sc-6jf9e4-0 cJrlLI"
      color="light.100"
      data-testid="ds-text"
    >
      Click Me
    </div>
  </button>
</div>
`;

exports[`<Button /> icon renders <Button /> with Info icon  1`] = `
<div>
  <button
    class="Buttonweb__StyledButton-gxhxmy-0 cXFeAh Spaceweb__Space-njatzl-0 jGLucJ Sizeweb__Size-sc-10bl685-0 bPNzzG Flexweb__Flex-ni2cjg-0 dfbkqx"
    color="primary"
    data-testid="ds-button"
  >
    <svg
      data-testid="ds-icon"
      fill="rgba(255, 255, 255, 1.0)"
      height="20"
      viewBox="0 0 16 16"
      width="20"
    >
      <mask
        height="14"
        id="prefix__a"
        maskUnits="userSpaceOnUse"
        width="14"
        x="1"
        y="1"
      >
        <g
          clip-path="url(#prefix__clip0)"
          fill="#fff"
        >
          <path
            d="M7.995 7.44c.307 0 .555.248.555.555v2.222a.555.555 0 11-1.11 0V7.995c0-.307.248-.555.555-.555zM7.995 5.218a.555.555 0 100 1.11h.006a.555.555 0 100-1.11h-.006z"
          />
          <path
            clip-rule="evenodd"
            d="M1.885 7.995a6.11 6.11 0 1112.22 0 6.11 6.11 0 01-12.22 0zm6.11-4.999a4.999 4.999 0 100 9.998 4.999 4.999 0 000-9.998z"
            fill-rule="evenodd"
          />
        </g>
      </mask>
      <g
        mask="url(#prefix__a)"
      >
        <path
          d="M0 0h16v16H0z"
          fill="#162F56"
          fill-opacity="0.38"
        />
      </g>
      <defs>
        <clippath
          id="prefix__clip0"
        >
          <path
            d="M1.33 1.33h13.33v13.33H1.33V1.33z"
            fill="#fff"
          />
        </clippath>
      </defs>
    </svg>
    <div
      class="Viewweb__View-sc-1cphyjf-0 lhSUkp Spaceweb__Space-njatzl-0 fIWTfg"
    />
    <div
      class="Textweb__Text-sc-6jf9e4-0 cJrlLI"
      color="light.100"
      data-testid="ds-text"
    >
      Click Me
    </div>
  </button>
</div>
`;

exports[`<Button /> icon renders a large Primary <Button /> with right icon 1`] = `
<div>
  <button
    class="Buttonweb__StyledButton-gxhxmy-0 cXFeAh Spaceweb__Space-njatzl-0 jGLucJ Sizeweb__Size-sc-10bl685-0 eFHgXX Flexweb__Flex-ni2cjg-0 dfbkqx"
    color="primary"
    data-testid="ds-button"
  >
    <svg
      data-testid="ds-icon"
      fill="rgba(255, 255, 255, 1.0)"
      height="20"
      viewBox="0 0 16 16"
      width="20"
    >
      <mask
        height="14"
        id="prefix__a"
        maskUnits="userSpaceOnUse"
        width="14"
        x="1"
        y="1"
      >
        <g
          clip-path="url(#prefix__clip0)"
          fill="#fff"
        >
          <path
            d="M7.995 7.44c.307 0 .555.248.555.555v2.222a.555.555 0 11-1.11 0V7.995c0-.307.248-.555.555-.555zM7.995 5.218a.555.555 0 100 1.11h.006a.555.555 0 100-1.11h-.006z"
          />
          <path
            clip-rule="evenodd"
            d="M1.885 7.995a6.11 6.11 0 1112.22 0 6.11 6.11 0 01-12.22 0zm6.11-4.999a4.999 4.999 0 100 9.998 4.999 4.999 0 000-9.998z"
            fill-rule="evenodd"
          />
        </g>
      </mask>
      <g
        mask="url(#prefix__a)"
      >
        <path
          d="M0 0h16v16H0z"
          fill="#162F56"
          fill-opacity="0.38"
        />
      </g>
      <defs>
        <clippath
          id="prefix__clip0"
        >
          <path
            d="M1.33 1.33h13.33v13.33H1.33V1.33z"
            fill="#fff"
          />
        </clippath>
      </defs>
    </svg>
    <div
      class="Viewweb__View-sc-1cphyjf-0 lhSUkp Spaceweb__Space-njatzl-0 fIWTfg"
    />
    <div
      class="Textweb__Text-sc-6jf9e4-0 cJrlLI"
      color="light.100"
      data-testid="ds-text"
    >
      Click Me
    </div>
  </button>
</div>
`;

exports[`<Button /> icon renders an icon only <Button /> 1`] = `
<div>
  <button
    class="Buttonweb__StyledButton-gxhxmy-0 cXFeAh Spaceweb__Space-njatzl-0 jGLucJ Sizeweb__Size-sc-10bl685-0 bPNzzG Flexweb__Flex-ni2cjg-0 dfbkqx"
    color="primary"
    data-testid="ds-button"
  >
    <div
      class="Textweb__Text-sc-6jf9e4-0 cJrlLI"
      color="light.100"
      data-testid="ds-text"
    >
      Click Me
    </div>
  </button>
</div>
`;

exports[`<Button /> size renders a large Primary <Button /> 1`] = `
<div>
  <button
    class="Buttonweb__StyledButton-gxhxmy-0 cXFeAh Spaceweb__Space-njatzl-0 jGLucJ Sizeweb__Size-sc-10bl685-0 eFHgXX Flexweb__Flex-ni2cjg-0 dfbkqx"
    color="primary"
    data-testid="ds-button"
  >
    <div
      class="Textweb__Text-sc-6jf9e4-0 cJrlLI"
      color="light.100"
      data-testid="ds-text"
    >
      Click Me
    </div>
  </button>
</div>
`;

exports[`<Button /> size renders a small Primary <Button /> 1`] = `
<div>
  <button
    class="Buttonweb__StyledButton-gxhxmy-0 cXFeAh Spaceweb__Space-njatzl-0 gkgBsY Sizeweb__Size-sc-10bl685-0 cwUXCl Flexweb__Flex-ni2cjg-0 dfbkqx"
    color="primary"
    data-testid="ds-button"
  >
    <div
      class="Textweb__Text-sc-6jf9e4-0 iWtrZV"
      color="light.100"
      data-testid="ds-text"
    >
      Click Me
    </div>
  </button>
</div>
`;

exports[`<Button /> size renders a small Primary <Button /> 2`] = `
<div>
  <button
    class="Buttonweb__StyledButton-gxhxmy-0 cXFeAh Spaceweb__Space-njatzl-0 gkgBsY Sizeweb__Size-sc-10bl685-0 cwUXCl Flexweb__Flex-ni2cjg-0 dfbkqx"
    color="primary"
    data-testid="ds-button"
  >
    <div
      class="Textweb__Text-sc-6jf9e4-0 iWtrZV"
      color="light.100"
      data-testid="ds-text"
    >
      Click Me
    </div>
  </button>
</div>
`;

exports[`<Button /> size renders xsmall primary <Button /> 1`] = `
<div>
  <button
    class="Buttonweb__StyledButton-gxhxmy-0 cXFeAh Spaceweb__Space-njatzl-0 hqmUpj Sizeweb__Size-sc-10bl685-0 yoKtN Flexweb__Flex-ni2cjg-0 dfbkqx"
    color="primary"
    data-testid="ds-button"
  >
    <div
      class="Textweb__Text-sc-6jf9e4-0 dptpNs"
      color="light.100"
      data-testid="ds-text"
    >
      CLICK ME
    </div>
  </button>
</div>
`;

exports[`<Button /> variant Renders a large Tertiary <Button /> 1`] = `
<div>
  <button
    class="Buttonweb__StyledButton-gxhxmy-0 bMjOWC Spaceweb__Space-njatzl-0 jGLucJ Sizeweb__Size-sc-10bl685-0 eFHgXX Flexweb__Flex-ni2cjg-0 dfbkqx"
    color="primary"
    data-testid="ds-button"
  >
    <div
      class="Textweb__Text-sc-6jf9e4-0 WxbKL"
      color="primary.800"
      data-testid="ds-text"
    >
      Click Me
    </div>
  </button>
</div>
`;

exports[`<Button /> variant renders a large Secondary <Button /> 1`] = `
<div>
  <button
    class="Buttonweb__StyledButton-gxhxmy-0 gyzoWG Spaceweb__Space-njatzl-0 jGLucJ Sizeweb__Size-sc-10bl685-0 eFHgXX Flexweb__Flex-ni2cjg-0 dfbkqx"
    color="primary"
    data-testid="ds-button"
  >
    <div
      class="Textweb__Text-sc-6jf9e4-0 WxbKL"
      color="primary.800"
      data-testid="ds-text"
    >
      Click Me
    </div>
  </button>
</div>
`;

exports[`<Button /> variant renders a medium(Default) Secondary <Button /> 1`] = `
<div>
  <button
    class="Buttonweb__StyledButton-gxhxmy-0 gyzoWG Spaceweb__Space-njatzl-0 jGLucJ Sizeweb__Size-sc-10bl685-0 bPNzzG Flexweb__Flex-ni2cjg-0 dfbkqx"
    color="primary"
    data-testid="ds-button"
  >
    <div
      class="Textweb__Text-sc-6jf9e4-0 WxbKL"
      color="primary.800"
      data-testid="ds-text"
    >
      Click Me
    </div>
  </button>
</div>
`;

exports[`<Button /> variant renders a medium(Default) Tertiary <Button /> 1`] = `
<div>
  <button
    class="Buttonweb__StyledButton-gxhxmy-0 bMjOWC Spaceweb__Space-njatzl-0 jGLucJ Sizeweb__Size-sc-10bl685-0 bPNzzG Flexweb__Flex-ni2cjg-0 dfbkqx"
    color="primary"
    data-testid="ds-button"
  >
    <div
      class="Textweb__Text-sc-6jf9e4-0 WxbKL"
      color="primary.800"
      data-testid="ds-text"
    >
      Click Me
    </div>
  </button>
</div>
`;

exports[`<Button /> variant renders a small Primary <Button /> 1`] = `
<div>
  <button
    class="Buttonweb__StyledButton-gxhxmy-0 cXFeAh Spaceweb__Space-njatzl-0 gkgBsY Sizeweb__Size-sc-10bl685-0 cwUXCl Flexweb__Flex-ni2cjg-0 dfbkqx"
    color="primary"
    data-testid="ds-button"
  >
    <div
      class="Textweb__Text-sc-6jf9e4-0 iWtrZV"
      color="light.100"
      data-testid="ds-text"
    >
      Click Me
    </div>
  </button>
</div>
`;

exports[`<Button /> variant renders a small Tertiary <Button /> 1`] = `
<div>
  <button
    class="Buttonweb__StyledButton-gxhxmy-0 bMjOWC Spaceweb__Space-njatzl-0 gkgBsY Sizeweb__Size-sc-10bl685-0 cwUXCl Flexweb__Flex-ni2cjg-0 dfbkqx"
    color="primary"
    data-testid="ds-button"
  >
    <div
      class="Textweb__Text-sc-6jf9e4-0 fZRaOO"
      color="primary.800"
      data-testid="ds-text"
    >
      Click Me
    </div>
  </button>
</div>
`;
